gmso/core/element.py

```diff
@@ -35,6 +35,13 @@ def __repr__(self):
                                                                       self.name, self.symbol,
                                                                       self.atomic_number,
                                                                       self.mass)
+    def __eq__(self, other):
+        return hash(self) == hash(other)
+
+    def __hash__(self):
+        return hash((self.name, self.symbol,
+                     self.atomic_number,
+                float(self.mass.to('amu').value)))
```

Comment:
Maybe we should use the `unyt_to_hashable` function for this just to be consistent. 

https://github.com/mosdef-hub/gmso/blob/master/gmso/utils/misc.py